bvm/ballerina-core/src/main/java/org/ballerinalang/model/values/BDecimal.java

```diff
@@ -58,11 +59,80 @@ public BDecimal(BigDecimal value) {
         }
     }
 
+    public BDecimal(String value) {
+        // Check whether the number provided is a hexadecimal value.
+        if (isHexValueString(value)) {
+            this.value = hexToDecimalFloatingPointNumber(value);
+        } else {
+            this.value = new BigDecimal(value, MathContext.DECIMAL128);
+        }
+        if (!this.booleanValue()) {
+            this.valueKind = DecimalValueKind.ZERO;
+        }
+    }
+
     public BDecimal(String value, DecimalValueKind valueKind) {
-        this.value = new BigDecimal(value, MathContext.DECIMAL128);
+        this(value);
         this.valueKind = valueKind;
     }
 
+    private boolean isHexValueString(String value) {
+        String upperCaseValue = value.toUpperCase();
+        return upperCaseValue.startsWith("0X") || upperCaseValue.startsWith("-0X");
+    }
+
+    /**
+     * Method used to convert the hexadecimal number to decimal floating point number.
+     * BigDecimal does not support hexadecimal numbers. Hence, we need to convert the hexadecimal number to a
+     * decimal floating point number before passing the string value to the BigDecimal constructor.
+     *
+     * @param value Hexadecimal string value that needs to be converted.
+     * @return BigDecimal corresponds to the hexadecimal number provided.
+     */
+    private BigDecimal hexToDecimalFloatingPointNumber(String value) {
```

Comment:
This should be a static method since you are calling from the constructor.

Reply:
Commit b47d555 addresses this.